test/forge/FixedRateIrmTest.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../../src/FixedRateIrm.sol";
+
+import "../../lib/forge-std/src/Test.sol";
+
+contract FixedRateIrmTest is Test {
+    using MarketParamsLib for MarketParams;
+
+    event SetBorrowRate(Id indexed id, uint256 newBorrowRate);
+
+    FixedRateIrm public fixedRateIrm;
+
+    function setUp() external {
+        fixedRateIrm = new FixedRateIrm();
+    }
+
+    function testSetBorrowRate(Id id, uint256 newBorrowRate) external {
+        vm.assume(newBorrowRate != 0);
+
+        fixedRateIrm.setBorrowRate(id, newBorrowRate);
+        assertEq(fixedRateIrm._borrowRate(id), newBorrowRate);
+    }
+
+    function testSetBorrowRateEvent(Id id, uint256 newBorrowRate) external {
+        vm.assume(newBorrowRate != 0);
+
+        vm.expectEmit(true, true, true, true, address(fixedRateIrm));
+        emit SetBorrowRate(id, newBorrowRate);
+        fixedRateIrm.setBorrowRate(id, newBorrowRate);
+    }
+
+    function testSetBorrowRateAlreadySet(Id id, uint256 newBorrowRate1, uint256 newBorrowRate2) external {
+        vm.assume(newBorrowRate1 != 0);
+        vm.assume(newBorrowRate2 != 0);
+        fixedRateIrm.setBorrowRate(id, newBorrowRate1);
+        vm.expectRevert("rate set");
```

Comment:
you can import errors from the contract maybe

Reply:
yep done